src/components/OSCALControl.js

```diff
@@ -41,6 +41,32 @@ const useStyles = makeStyles(() => ({
   OSCALControlStatus: (props) => getControlStatusCss(props),
 }));
 
+/**
+ * Retrieve the set-parameters when contained in the top-level implemented requirement statement
+ *
+ * @param {Object} implReqStatements
+ * @param {String} componentId
+ * @returns An array of set-parameters
+ */
+function getImplReqSetParameters(implReqStatements, componentId) {
+  // Get the top-level implemented requirement statement
+  let topLevelImplReqStatement = null;
+  if (implReqStatements) {
+    // NOTE: The top level statement ends with "_smt"
+    topLevelImplReqStatement = implReqStatements.find((statement) =>
+      statement["statement-id"].endsWith("_smt")
+    );
+  }
+  // NOTE: The top-level by-component has a matching uuid to the componentId
+  let topLevelByComp = null;
+  if (topLevelImplReqStatement?.["by-components"]) {
+    topLevelByComp = topLevelImplReqStatement["by-components"].find(
+      (byComp) => byComp["component-uuid"] === componentId
+    );
+  }
+  return topLevelByComp?.["set-parameters"] || null;
```

Comment:
I could be wrong here, but is it possible to make this into just a pipe of `find().find()`? It seems like there could be a cleaner way of doing this.